### ISSUE TITLE
Data check actions: infer missing datetime values

### ISSUE THREAD

**Issue by dsherry** [alteryx/evalml#2022]
A few starting options for missing datetime values:
* Drop all rows with nans
* Backfill with next non-nan value
* Forward-fill with previous non-nan value
* For nans between two non-nan values, fill with linearly-spaced datetimes. Drop rows before the 1st non-nan value and after the last non-nan value.

**Comment by freddyaboulton** [comment]
I think the recommendation should depend on what kind of dataset it is. The filling operations only makes sense where there is a defined relationship between the rows, i.e. time series regression problems. It may not make sense, for example, in a taxi cab dataset where I'm trying to predict the length of the trip based on the start, end location and the time of day. What's the next or previous date in that case is not defined.

Another solution I was thinking about is making our `DateTimeFeaturizer` resistant to null values by making it return -1 or "missing" as the features it creates. Since time series regression is still in beta,  I think approach 1 (drop rows) or updating the `DateTimeFeaturizer` should be given priority in this first pass.

**Comment by dsherry** [comment]
@freddyaboulton I agree! Different datasets will require different strategies. That's what is helpful about the data check actions API -- the user gets to decide which action to take. As long as we support actions which are useful in different situations, we should be in good shape.

Once #2070 is merged, our code will be able to tell the difference between the datetime index for timeseries and a general datetime feature, so that is helpful.

I agree "drop all rows with nan" is the right starting place.
